service/mongodb/store.go

```diff
@@ -0,0 +1,155 @@
+package mongodb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/meateam/fav-service/service"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+)
+
+const (
+	// MongoObjectIDField is the default mongodb unique key.
+	MongoObjectIDField = "_id"
+
+	// FavoriteCollectionName is the name of the favorites collection.
+	FavoriteCollectionName = "favorite"	
+
+	// FavoriteBSONFileIDField is the name of the fileID field in BSON.
+	FavoriteBSONFileIDField = "fileID"
+
+	// FavoriteBSONUserIDField is the name of the userID field in BSON.
+	FavoriteBSONUserIDField = "userID"
+)
+
+// MongoStore holds the mongodb database and implements Store interface.
+type MongoStore struct {
+	DB *mongo.Database
+}
+
+// newMongoStore returns a new store.
+func newMongoStore(db *mongo.Database) (MongoStore, error) {
+	collection := db.Collection(FavoriteCollectionName)
+	indexes := collection.Indexes()
+	indexModel := mongo.IndexModel{
+		Keys: bson.D{
+			bson.E{
+				Key:   FavoriteBSONFileIDField,
+				Value: 1,
+			},
+			bson.E{
+				Key:   FavoriteBSONUserIDField,
+				Value: 1,
+			},
+		},
+		Options: options.Index().SetUnique(true),
+	}
```

Comment:
[Clickme](https://www.youtube.com/watch?v=dO1rMeYnOmM)
We already talked about indexing here

Reply:
I would like to talk about it again. I think this part makes sense.